### PR TITLE
Update mapdraw control and add map search on projectCreate

### PR DESCRIPTION
**Pull request by yunjuc**
Implement #3353 and add map search.
- On step 1, add new draw control to activate draw on the map. Support toggle control between step 1 and 2.
- Add search to the map.

### CODE REVIEW

**Comment by yunjuc** [comment]
For the conflict file, I only add `@mapbox/mapbox-gl-geocoder` in this PR.

There is a warning when frontend server starts- **React Hook useLayoutEffect has a missing dependency: 'mapObj.draw'** It's intentional because watching **mapObj.draw** will cause the draw control being appended multiple times.

File: frontend/src/components/projectCreate/projectCreationMap.js [hunk 1]
```diff
@@ -61,56 +63,48 @@ const ProjectCreationMap = ({ mapObj, setMapObj, metadata, updateMetadata, step,
...
+          center: [0, 0],
+          zoom: 1,
+          attributionControl: false,
+        })
+          .addControl(new mapboxgl.AttributionControl({ compact: false }))
+          .addControl(new MapboxLanguage({ defaultLanguage: locale.substr(0, 2) || 'en' }))
+          .addControl(new mapboxgl.ScaleControl({ unit: 'metric' }))
+          .addControl(new MapboxGeocoder({
+              accessToken: MAPBOX_TOKEN,
+              mapboxgl: mapboxgl,
```
Comment (willemarcel) [inline frontend/src/components/projectCreate/projectCreationMap.js:81]: Add more 2 lines to the Geocoder config:
marker: false,
language: locale.substr(0, 2) || 'en'

**Comment by willemarcel** [comment]
Thanks, @yunjuc! I'm sorry I couldn't review it earlier. I added a comment with an improvement and I also rebased it with develop on https://github.com/hotosm/tasking-manager/tree/map-draw

The only suggestion I have is to style the draw button in order to call more attention. As the users are used to have a button on the right, they may not notice the new button. Maybe making it bigger or including a text would be a good solution

**Comment by yunjuc** [comment]
I enlarged the control icons and moved them to the left-hand side so they are closer to the instructions. 
Screen Shot 2020-12-21 at 6 33 25 PM

Two questions:
1. Is **_extra.scss** the right place to add custom style?
2. Since you rebased the code to a new branch, should I push the code here, or open a new PR and push the update there?

**Comment by willemarcel** [comment]
>     1. Is **_extra.scss** the right place to add custom style?

Yes, it's.

>     2. Since you rebased the code to a new branch, should I push the code here, or open a new PR and push the update there?

I think, the best here is that you download the new branch: `git checkout -b new-map-draw origin/map-draw`, considering that your origin is the hotosm repository.

Then checkout the branch you used to open this PR and execute `git rebase new-map-draw`. If it doesn't work, let me know

**Comment by willemarcel** [comment]
Hi @yunjuc ! Thanks a lot for your collaboration. I used some of your code on https://github.com/hotosm/tasking-manager/pull/4089

**Pull request closed without merging**